Column Intervals, board start and end edges, column background

LisovyiDK opened the pull request:
Hi! I've added the possibility to set spaces for columns (intervals between columns and board start & end edges). Also, I've added the possibility to set items' recycler background color.

woxblom commented:
Hey,

Thanks for the pull request :) I got some questions however.

1. When adding board edges the first and last column will have less width than all other columns, is this intended? Feels a bit weird to me.

2. When adding column intervals the columns are no longer centered when swiping horizontally and snapping. Is this intended? Also feels a bit weird.

3. The ItemsParameters feature you added, is it more of a convenience thing? As you can both set fixed with and background color of the RecyclerView already.

Cheers

LisovyiDK commented:
Thank you for drawing my attention to these bugs. I have fixed bug **No.1** and **No.2** also I fixed updating edges on deleting columns but still not pushed. I just want to agree with you on the solution of **No.3**.

You are right that I can set fixed size and background color to the instance of RecyclerView received from the addColumn method. But is it a good solution to back the recyclerView instance? Seems you have set drag listeners and callbacks that shouldn't be overridden. 

So I want to indicate the old methods where had to be passed fixed size and layout manager as deprecated and add the new methods which will not back the recyclerView and where all necessary settings will be passed via ItemsParameters. What do you think about this?

woxblom commented:
Hey,

Yeah I can agree with that.
I would like to change the name to ColumnProperties however since it is the column (recyclerview) you are configuring.
You wanna add the adapter, header and columnDragView in the ColumnProperties as well or?

You can fix the deprecation and add new method if you like, or I do it after the merge.

LisovyiDK commented:
Yes, if it is named ColumnProperties it will be better to add adapter, header and columnDragView there as well. I'll also add the whole column background parameter.

LisovyiDK commented:
Sorry for the delay. 
I've made the following changes:
- fixed column intervals
- fixed board edges
- renamed ItemsParameters to ColumnProperties
- added column header and column drug view to the ColumnProperties
- added the possibility to set column background and items list background
- added the possibility to set itemDecorator via  ColumnProperties
- indicated old add and insert methods as deprecated.

Please look the last commit.

woxblom commented on library/src/main/java/com/woxthebox/draglistview/ColumnProperties.java:
```diff
@@ -0,0 +1,218 @@
+                    mColumnBackgroundColor,
+                    mItemsSectionBackgroundColor,
+                    mHeader,
+                    mColumnDragView);
```
mHeader and mColumnDragView is added in the wrong order

LisovyiDK replied:
Changed

woxblom commented on library/src/main/java/com/woxthebox/draglistview/ColumnProperties.java:
```diff
@@ -0,0 +1,218 @@
+         * @return instance of the {@link Builder}
+         */
+        public Builder setHasFixedItemSize(boolean hasFixedItemSize) {
+            this.mHasFixedItemSize = hasFixedItemSize;
```
remove "this."

LisovyiDK replied:
Removed

woxblom commented on library/src/main/java/com/woxthebox/draglistview/ColumnProperties.java:
```diff
@@ -0,0 +1,218 @@
+        return mItemDecorations;
+    }
+
+    boolean isHasFixedItemSize() {
```
change to hasFixedItemSize()

LisovyiDK replied:
Changed

woxblom commented on library/src/main/java/com/woxthebox/draglistview/ColumnProperties.java:
```diff
@@ -0,0 +1,218 @@
+        private ArrayList<RecyclerView.ItemDecoration> mItemDecoration = new ArrayList<>();
+        private boolean mHasFixedItemSize = false;
+        private int mColumnBackgroundColor = Color.TRANSPARENT;
+        private int mItemsSectionBackgroundColor = Color.TRANSPARENT;
```
Curious about this one, was this added to be able to have a different color of the header background? Since if no header is added mItemsSectionBackgroundColor will always overwrite the column color. Is it more logical to only have mColumnBackgroundColor and then you control the header background color from the actual header view that you choose to add yourself?

LisovyiDK replied:
I faced the case when I needed to have a transparent header background and colored background of the items list. I prefer to leave mItemsSectionBackgroundColor, but in my opinion, if we gather all the properties of a column to one file we should add the possibility to set the whole column color. Also, we can add a frame for the header and set colors to the header and to the items separately. What do you think about this?

woxblom replied:
Ah I see, let's leave it like this.

woxblom commented on sample/src/main/res/values/dimens.xml:
```diff
@@ -3,6 +3,6 @@
 
     <!-- Margins -->
     <dimen name="columnIntervals">8dp</dimen>
```
I feel like columnSpacing is a better name for this, seems more commonly used for this kind of behaviour

LisovyiDK replied:
Renamed all Intervals to Spaces

woxblom commented on library/src/main/java/com/woxthebox/draglistview/ColumnProperties.java:
```diff
@@ -0,0 +1,218 @@
+package com.woxthebox.draglistview;
```
Add Apache License in top as in other files. Also add in ViewUtils file

LisovyiDK replied:
Added

woxblom commented on sample/src/main/res/values/colors.xml:
```diff
@@ -14,6 +14,7 @@
      limitations under the License.
```
Add Apache copyright in top as in other xml files

LisovyiDK replied:
Added

woxblom commented on library/src/main/java/com/woxthebox/draglistview/ColumnProperties.java:
```diff
@@ -0,0 +1,218 @@
+         *
+         * @return instance of the {@link Builder}
+         */
+        public Builder setColumnDrugView(@Nullable View columnDragView) {
```
setColumnDragView instead of Drug :)

LisovyiDK replied:
Corrected :)

woxblom commented on library/src/main/java/com/woxthebox/draglistview/ViewUtils.java:
```diff
@@ -0,0 +1,16 @@
+package com.woxthebox.draglistview;
```
Add Apache copyright in top

LisovyiDK replied:
Added

woxblom commented:
I have now merge and released a 1.7.0 version of the lib.
Thanks for your contribution!
